rust/rpc/src/lib.rs

```diff
@@ -249,6 +256,7 @@ impl<W: Write + Send> RpcLoop<W> {
             });
 
             loop {
+                let read_result_trace = xi_trace::trace_block("read & process result", &["rpc"]);
```

Comment:
I think rather than explicitly dropping these my preference would be to add a new block for the scope of the trace.